src/etl/v2g/functional_predictions/vep.py

```diff
@@ -0,0 +1,201 @@
+"""The variant annotation dataset contains information about the impact of a variant on a transcript or protein. These can be mapped to genes allowing us to establish significant relationships between variants and genes."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pyspark.sql.functions as f
+
+from etl.common.spark_helpers import get_record_with_maximum_value
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame
+
+    from etl.common.ETLSession import ETLSession
+
+
+def main(
+    etl: ETLSession,
+    variant_index_path: str,
+    variant_annotation_path: str,
+    variant_consequence_lut_path: str,
+) -> tuple[DataFrame, ...]:
+    """Extracts variant to gene assignments for the variants included in the index and the features predicted by VEP.
+
+    Args:
+        etl (ETLSession): ETL session,
+        variant_index_path (str): The path to the OTG variant index
+        variant_annotation_path (str): The path to the variant annotation file
+        variant_consequence_lut_path (str): The path to the LUT between the functional consequences and their assigned V2G score
+
+    Returns:
+        DataFrame: variant to gene assignments from VEP
+    """
+    # function that takes the index, gets the vep data and parses it
+    etl.logger.info("Loading variant index")
+    variant_consequence_lut = read_consequence_lut(etl, variant_consequence_lut_path)
+    va_with_vep = (
+        etl.spark.read.parquet(variant_annotation_path)
```

Comment:
better to use `read_parquet(variant_annotation_path, "variant_annotation.json")` as we already have it

Reply:
I forgot to add this, thanks :) Changed in https://github.com/opentargets/genetics_etl_python/commit/ebe6717b1207d1629685a71421b4cbce3c4b590b